src/components/NewCommentForm.tsx

```diff
@@ -10,28 +89,36 @@
 
         <div className="control has-icons-left has-icons-right">
           <input
+            value={authorName}
+            onChange={handleAuthorNameChange}
             type="text"
             name="name"
             id="comment-author-name"
             placeholder="Name Surname"
-            className="input is-danger"
+            className={classNames('input', {
+              'is-danger': authorNameError.length > 0,
+            })}
           />
 
           <span className="icon is-small is-left">
             <i className="fas fa-user" />
           </span>
 
           <span
-            className="icon is-small is-right has-text-danger"
+            className={classNames('icon is-small is-right ', {
+              'has-text-danger': authorNameError.length > 0,
+            })}
             data-cy="ErrorIcon"
           >
             <i className="fas fa-exclamation-triangle" />
           </span>
         </div>
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Name is required
-        </p>
+        {authorNameError && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            Name is required
+          </p>
```

Comment:
Instead of hardcoding the error message 'Name is required', consider dynamically displaying the `authorNameError` message. This will make the error handling more flexible and informative.